CRM/Contact/BAO/Query.php

```diff
@@ -805,42 +822,9 @@ public function selectClause($apiEntity = NULL) {
                 $this->_element['provider_id'] = 1;
               }
 
-              if ($tName == 'contact') {
+              if ($tName == 'contact' && $fieldName == 'organization_name') {
                 // special case, when current employer is set for Individual contact
-                if ($fieldName == 'organization_name') {
-                  $this->_select[$name] = "IF ( contact_a.contact_type = 'Individual', NULL, contact_a.organization_name ) as organization_name";
-                }
-                elseif ($fieldName != 'id') {
```

Comment:
move all this stuff out of the 'contact-table-specific' & make it just 'metadata-based'